File: classes/parte-7-paradigmas/111-herencia-composicion-y-polimorfismo/implementaciones/c/main.c
#include <stdio.h>
#include <string.h>

int main(void) {
    char tipo[32];
    if (scanf("%31s", tipo) != 1) return 1;
    const char *sonido;
    if (strcmp(tipo, "perro") == 0) sonido = "guau";
    else if (strcmp(tipo, "gato") == 0) sonido = "miau";
    else sonido = "muu";
    printf("sonido=%s\n", sonido);
    return 0;
}

File: classes/parte-7-paradigmas/111-herencia-composicion-y-polimorfismo/implementaciones/java/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    interface Animal { String sonido(); }
    static class Perro implements Animal { public String sonido() { return "guau"; } }
    static class Gato implements Animal { public String sonido() { return "miau"; } }
    static class Vaca implements Animal { public String sonido() { return "muu"; } }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String tipo = br.readLine().trim();
        Animal a;
        switch (tipo) {
            case "perro": a = new Perro(); break;
            case "gato": a = new Gato(); break;
            default: a = new Vaca();
        }
        System.out.println("sonido=" + a.sonido());
    }
}
